### PR TITLE
Separate suffixes for date/month/year in short date + make delimiters optional

### LINKED ISSUE #189: Feature Request: year/month/day specific delimiters/suffixes
First off, I'd like to say that I've been using CaretTab for about a year and a half now, and it's better than pretty much any other New Tab extension I've used. (in my opinion)

Anyway, I'm requesting being able to use different delimiters/suffixes for the year, month and day on the date.

That way, I can use the Japanese date notation system. (ex: 2008年12月31日)

Thanks in advance!

### PR DESCRIPTION
**Pull request by coffeeologist**
## What?
These changes allow

1. users to set suffixes for the short date format and 
2. turn off delimiters for short date format
 
Suffixes, different from delimiters, appear immediately after the corresponding numeric value (whether that's date, month, or year), and can be separately set to different strings.

Fixes #189

## Why?
This is a requested enhancement feature and a currently open issue: #189. Plus, it better enhances the user experience for non-English speakers. 

## How?

- RE: turning delimiters optional. In the original usage example, there are no delimiters, thus they are turned into optional features that can be toggled on/off. There is now a `enableDelimiter` setting inside `settings.ts` to keep track of this. The html page for date also checks whether delimiters are enabled before displaying them. 
- Suffixes are implemented in the same way as delimiters, though 1 `enableSuffix` setting controls the visibility of all 3 suffixes (`dateSuffix`, `monthSuffix`, and `yearSuffix`). It has the same logic of display, dimming, and change in setting as delimiters code-wise. 

## Testing?
- Successful local dev build
- Manual testing through clicking and toggling

## Screenshots
As shown, users can set their own suffixes for each of year, month, and date. These suffixes can be dimmed as well. The screenshot displays the usage example outlined in #189.

Delimiters can coexist with suffixes, so that users can use them to better separate the numbers with - or / , or just set it as a space.

## Anything Else?
- The current character cap on suffixes is 3. More research may be needed to decide on a better cap to encompass all language use cases. 
- Translation is needed for the numerous blurbs that appears in the "date" panel:
    - suffix
    - dimDateDlelimiter
    - dimDateSuffix
    - enableDelimiter
    - enableSuffix

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,13 +45,18 @@ You can install CaretTab through the browser web stores
 Submit a Pull Request in GitHub if you would like to contribute. Always welcome help with translations especially :)
 
 ## Build
-Run `npm start` for a dev build. You can then run `npm run start:chrome` to start an instance of the dev build in Chrome.
+### First time installation:
+1. Make sure you have [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) installed 
+2. Run `npm install` for packages necessary to build the project
+3. Run `npm install web-ext` to be able to start a dev build instance in Chrome
```
Comment (bluecaret) [inline README.md:51]: Should be global install or at least dev dependency

**Review by bluecaret — CHANGES_REQUESTED** [review]
Please separate the work on the suffixes from the work done in other PR's so I can review just this change. I don't believe there is any dependency on the other PRs for this. Thanks

**Comment by coffeeologist** [comment]
So sorry about this! I didn't notice they were all jumbled up. I ended up resetting the branch back to the current `bluecaret/carettabl/master` latest commit and added the changes regarding suffixes back in. Functionality wise it should have no change. However, this PR will not include the readme and PR template that the other PRs have.

**Comment by bluecaret** [comment]
I want to apologize for ignoring these for SOOO long! At the time I was too busy with my day job and life that I just never got back to this PR. Since then some things have changed in the code that make your PRs unable to be merged. However, I did implement the features, copy/pasting your code where I could. So rest assured, your effort was not for nothing, consider your three PRs, #257 #258 #260 all to be completed! Due to the merge issues I'm actually closing these PRs though. Hope that makes sense and thank you for the help!

**Pull request closed without merging**